src/main/java/apoc/util/Util.java

```diff
@@ -281,32 +282,68 @@ private static String handleRedirect(URLConnection con, String url) throws IOExc
 
     public static CountingInputStream openInputStream(String urlAddress, Map<String, Object> headers, String payload) throws IOException {
         StreamConnection sc;
+        InputStream stream;
+        if (urlAddress.contains(".zip") || urlAddress.contains(".tar") || urlAddress.contains(".tgz")) {
```

Comment:
shouldn't we do a `contains("!")` check first here
or at least check `zipFileName` for null before passing it into `getFileStreamIntoCompressedFile`